src/OpenTelemetry.Abstractions/Trace/ISpan.cs

```diff
@@ -117,5 +118,11 @@ public interface ISpan
         /// End the span.
         /// </summary>
         void End();
+
+        /// <summary>
+        /// End the span.
+        /// </summary>
+        /// <param name="endTimestamp">End UTC timestamp.</param>
+        void End(DateTime endTimestamp);
```

Comment:
DateTimeOffset is better suited for this use case

Reply:
Yes, you are right. So I guess what we should do, is to switch (at least) all public APIs to deal with `DateTimeOffset` rather than `DateTime`.

I'd say this should be done in a different PR - this will affect much more than Span start/end